### PR TITLE
[server] getRunData limitation

### PR DESCRIPTION
**Pull request by csordasmarton**
Previously `getRunData` function returned all the runs. It is possible that the user stored a lot of runs (`>1000`) so getting these data was too slow. For this reason we introduced a new limit/offset parameters for the getRunData function which limits the number of runs which will be returned to the user.

We've also introduced a new API function called `getRunCount` which will return the number of available runs. This can be filtered by a run filter parameter.

### CODE REVIEW

File: web/server/codechecker_server/api/report_server.py [hunk 1]
```diff
@@ -660,8 +681,15 @@ def __get_run_ids_to_query(session, cmp_data=None):
 
     @exc_to_thrift_reqfail
     @timeit
-    def getRunData(self, run_filter):
+    def getRunData(self, run_filter, limit, offset):
         self.__require_access()
+
+        max_query_limit = constants.MAX_QUERY_SIZE
```
Comment (gyorb) [inline web/server/codechecker_server/api/report_server.py]: Please check the query size with the newly introduced function in #2180

File: web/server/www/scripts/codecheckerviewer/ListOfRuns.js [hunk 2]
```diff
@@ -101,10 +103,67 @@ function (declare, dom, ItemFileWriteStore, topic, Dialog, Button,
...
+      var deferred = new Deferred();
+      deferred.total = query.total;
+
+      var that = this;
+      CC_SERVICE.getRunData(
+        query.runFilter,
+        CC_OBJECTS.MAX_QUERY_SIZE,
+        options.start,
+        function (runDataList) {
+          if (runDataList instanceof RequestFailed)
```
[the remark this answers is not in the document]
Comment (gyorb) [reply web/server/www/scripts/codecheckerviewer/ListOfRuns.js]: Is this just a codacy error or `RequestFailed` should be defined somewhere?

Comment (gyorb) [inline web/server/www/scripts/codecheckerviewer/ListOfRuns.js]: Shouldn't we use curly brackets here there is an else branch.
```suggestion
          if (runDataList instanceof RequestFailed){
          ...
          }
          else{
          ...
          }
```

[the remark this answers is not in the document]
Comment (csordasmarton) [reply web/server/www/scripts/codecheckerviewer/ListOfRuns.js]: This is a Codacy error, `RequestFailed` is defined here:
https://github.com/Ericsson/codechecker/blob/26dc78674f1f2243aea0695373a116ca37e41c88/web/api/shared.thrift#L17

**Pull request merged**